### DISCUSSION TITLE
Is it possible to add/rename color names?

### DISCUSSION THREAD

**Discussion by stefanobartoletti** [saadeghi/daisyui#1559] [Q&A] [answered]
Hi, first of all thank you for this library, it is complementing Tailwind perfectly.

I'd like to know if it is possible to customize the names of the colors.

In my setup, I want to have utilities like `bg-dark`, `bg-light`, `btn-dark`, `btn-light`, and I don't need, for example, `bg-accent`, `btn-accent` and so on.

Is there an easy way to customize these names, either by editing them or by adding more (I guess that unused tokens will be purged when bundling).

Thanks!

**Comment by saadeghi** [comment, marked as answer]
Thank you.  

1. The things you don't need like `btn-accent` as you mentioned, will not be generated if you never use the class name (like any other unused class name of course)

2. class names that are defined based on being `dark` or `light` (like `bg-dark` or `btn-light` you suggested) can not look good on both dark and light themes.  
Let's say we have a section with `bg-light` it would have a white background color. It's okay for a light theme but what should happen on a dark theme? We should either keep it white on both themes which is a bad idea or we should use Tailwind CSS' method and add a new class name with `dark:` prefix for every color name and set an alternative color value for them for the second theme (dark theme). Like `bg-light dark:bg-dark`.  
If that's how you need it, you can simple keep using [Tailwind CSS color names](https://tailwindcss.com/docs/customizing-colors) or [set your own color names](https://tailwindcss.com/docs/customizing-colors#adding-additional-colors) like `dark` color or `light` color.  
But daisyUI color names are using a different method. The colors are named after what were using them for. Not based on being dark or light or blue or red. For example we have `base-100`, `base-200` color names. These colors are the base color of page for blank background of sections. They can be dark in a dark theme or they can be light on a light theme. And then we have `base-content` which is the foreground color to be used on `base` colors. So if it's a dark theme (and `base-100` is a dark color), `base-content` will be a light color so the content would be readable.  
This is easier to use since we don't have to add `dark:` prefix for every color class name and set the alternative dark mode color for each element. we just use the semantic color name and it will work on light theme and dark theme and any additional themes.  
So the benefits of using semantic color names instead of constant color names like `bg-dark` or `bg-blue` would be:
  - 50% less class names for colors, smaller HTML size
  - smaller CSS size
  - you can have multiple themes (not just 1 dark theme and 1 light theme)
  - you can change 1 color value from the config and it changes everywhere
  - while writing HTML, you don't need to think about how the page will look on the other theme. If you use daisyUI color names, it will look good on both dark and light themes automatically
  - This is the the same standard way designers create a design system. When you make a design system for a specific app or brand, you don't use `blue`, `green`, `dark` or `light` you create names like `primary`, `secondary`, etc for brand colors and you create `background`, `foreground`, etc for the UI because the actual color value can get changed anytime.
Read more:  
- https://daisyui.com/docs/colors/
- https://daisyui.com/docs/themes/
- https://daisyui.com/theme-generator/

3. If you want to add new color names to daisyUI (that can be changed based on the theme – not names like `dark` or `light`), read this: https://github.com/saadeghi/daisyui/discussions/653#discussioncomment-2438428

Let me know if you have any questions

**Comment by stefanobartoletti** [comment]
Thank you, I see your reasoning and it makes sense. I will consider how to use the existing names to better suit my needs.

**Comment by stefanobartoletti** [comment]
I better read the link you gave, and it seems that adding new color names won't automatically create corresponding classes.

I mean, if I add a color `mycolor`, will classes like `bg-mycolor` or `btn-mycolor` be automatically generated?

**Reply by saadeghi** [reply]
No. component color variants don't get generated automatically for all color names because that would result many inaccessible color combinations. Each component that needs color variants, has a specific set of them as mentioned on the docs.  
You can add new color name as I mentioned, but if you need a custom variant for button, that's another thing. It really isn't something that can be automated.  
For example a button should have text color, background color, border color, and also background color and border color of hover state.  
If you want something like that, here's an example: https://play.tailwindcss.com/FTpzHUC9W1?file=css

But honestly I think you're trying to fight the design system instead of adapting it. [Existing color names](https://daisyui.com/docs/colors/) are more than enough to create almost any UI.
